src/rgw/driver/rados/rgw_sal_rados.cc

```diff
@@ -2904,7 +2904,8 @@ int RadosMultipartWriter::complete(size_t accounted_size, const std::string& eta
                        uint32_t flags)
 {
   return processor.complete(accounted_size, etag, mtime, set_mtime, attrs, delete_at,
-			    if_match, if_nomatch, user_data, zones_trace, canceled, rctx, flags);
+                            if_match, if_nomatch, user_data, zones_trace, canceled, rctx,
+                            flags & ~FLAG_LOG_OP);
```

Comment:
it makes sense to remove that flag for `MultipartWriter`s, and `RadosMultipartWriter` is a convenient place to do that, but it makes this change unnecessarily rados-specific. someday we'll support multisite sync for other sal backends too, and the behavior around multipart uploads will likely stay backend-agnostic

to make this more generic, `RGWPutObj::execute()` chould decide whether or not to pass FLAG_LOG_OP:
```diff
diff --git a/src/rgw/rgw_op.cc b/src/rgw/rgw_op.cc
index d1d332b0016..f5ed1089bfa 100644
--- a/src/rgw/rgw_op.cc
+++ b/src/rgw/rgw_op.cc
@@ -4587,12 +4587,16 @@ void RGWPutObj::execute(optional_yield y)
     emplace_attr(RGW_ATTR_OBJECT_RETENTION, std::move(obj_retention_bl));
   }
 
+  // don't track the individual parts of multipart uploads. they replicate in
+  // full after CompleteMultipart
+  const uint32_t complete_flags = multipart ? 0 : rgw::sal::FLAG_LOG_OP;
+
   tracepoint(rgw_op, processor_complete_enter, s->req_id.c_str());
   const req_context rctx{this, s->yield, s->trace.get()};
   op_ret = processor->complete(s->obj_size, etag, &mtime, real_time(), attrs,
                                (delete_at ? *delete_at : real_time()), if_match, if_nomatch,
                                (user_data.empty() ? nullptr : &user_data), nullptr, nullptr,
-                               rctx, rgw::sal::FLAG_LOG_OP);
+                               rctx, complete_flags);
   tracepoint(rgw_op, processor_complete_exit, s->req_id.c_str());
   if (op_ret < 0) {
     return;
```